### PR TITLE
feat(network_info_plus): add ipv6, gateway, broadcast and subnet mask information

### PR DESCRIPTION
**Pull request by v-apparence**
## Description

We added for both **iOS** and **Android** support for fetching information : 

- IPv6 address
- Broadcast address
- Gateway address
- Subnet mask

Linux, MacOS and Windows was not done in this PR and need help 👍 

## Related Issues

#206 
#132 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by mhadaily** [comment]
Looks like there are some issues with formating.

**Comment by mhadaily** [comment]
I have reviewed this code and LGTM. thank you for your PR. I feel this is a great addition to network_info plus. 

@jpnurmi  @miquelbeltran any comment or concerns?

**Comment by miquelbeltran** [comment]
changes look good to me! the formatting needs to be fixed, yes

**Comment by dim-apparence** [comment]
Hi, we can't get `melos run format`to work... we had on both computer (even after a fresh clean clone) this error:

```shell
Unhandled exception:
ProcessException: No such file or directory
# [...]
#0      _ProcessImpl._start (dart:io-patch/process_patch.dart:390:33)
#1      Process.start (dart:io-patch/process_patch.dart:36:20)
#2      ProcessRunner.runAndStream (package:flutter_plugin_tools/src/common/process_runner.dart:37:49)
#3      FormatCommand._formatCppAndObjectiveC (package:flutter_plugin_tools/src/format_command.dart:108:27)
<asynchronous suspension>
#4      FormatCommand.run (package:flutter_plugin_tools/src/format_command.dart:48:5)
<asynchronous suspension>
#5      CommandRunner.runCommand (package:args/command_runner.dart:196:13)
<asynchronous suspension>
#6      main.<anonymous closure> (package:flutter_plugin_tools/src/main.dart)
<asynchronous suspension>
```

Do you had some clues @mhadaily @miquelbeltran  ?
Thanks

**Comment by stuartmorgan** [comment]
@dim-apparence It looks like you don't have `clang-format` in your path. If you update to the version of `flutter_plugin_tools` that was just pushed, you should get a clear error message instead of that stack.

**Comment by v-apparence** [comment]
Thanks @stuartmorgan, `melos run format` works 👍

**Comment by mhadaily** [comment]
@v-apparence this is great so far. 

There is one thing that our code coverage deceased a bit I can see why it happens though. 

@miquelbeltran @jpnurmi any comment, if not, let's merge and publish this

**Comment by mhadaily** [comment]
@miquelbeltran @jpnurmi I am going to merge this PR, even though it has lower test coverage, do you have any objection to this?

**Comment by miquelbeltran** [comment]
@mhadaily  yes, let's go for it!

**Review by mhadaily — APPROVED** [review]

File: packages/network_info_plus/network_info_plus/ios/Classes/getgateway.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//  getgateway.h
+//  network_info_plus
+//
+//  Created by Dimitri Dessus on 08/07/2021.
+//
+
+#ifndef getgateway_h
+#define getgateway_h
+
+int getDefaultGateway(in_addr_t* addr);
```
Comment (ChristianEdwardPadilla) [inline packages/network_info_plus/network_info_plus/ios/Classes/getgateway.h:11]: Where is this type coming from?

**Pull request merged**